src/game_objects/game_object.cpp

```diff
@@ -64,16 +66,20 @@ void GameObject::draw_instanced(Shader *shader, SceneInfo &scene_info,
 // Runs Game Object's update loop
 void GameObject::update() {
 
-    update_this();
+    if(enabled)
+        update_this();
 
     for (GameObject *obj : children)
         obj->update();
 }
 
-void GameObject::update_state(float x, float z, float wx, float wz) {
-    transform.set_position(x, 0.0f, z);
-    direction = glm::vec3(wx, 0, wz);
-    transform.look_at(direction);
+void GameObject::update_state(float x, float z, float wx, float wz, bool enab) {
```

Comment:
I think we're starting to use too many parameters for this when it's supposed to be just an update function, but I guess it's fine since I don't really see a nicer way as of right now.